### DISCUSSION TITLE
a line on the screen at several points

### DISCUSSION THREAD

**Discussion by gekzakm** [zepp-health/zeppos-forum#305] [Developer Feedback]
Tell me a way to draw a curved line at several points. I'm trying to start programming, I'm new to this.I am writing for API_LEVEL 2.0. I have a desire to create a tracker based on a T-Rex2 watch

**Comment by silver-zepp** [comment]
We don't have an out-of-the-box support for curves. The best you can do is to use the [POLYLINE widget ](https://docs.zepp.com/docs/v2/reference/device-app-api/newAPI/ui/widget/GRADIENT_POLYLINE/)with your custom curve implementation.

**Reply by gekzakm** [reply]
Thanks for the quick response. I've tried using it. It is a pity that the height of this widget is only 150. There's no other way?

**Reply by silver-zepp** [reply]
Unfortunately, that's its current limitation. It's pretty memory intensive and only supports up to three widgets of the same type max. We will look into more optimal implementation in the future but for now, that's all we have.

**Reply by gekzakm** [reply]
do trainers using GPS also use this widget? T-rex2 is good for everyone, only a simple tracker is missing, with the ability to record a track and mark points along the way. Saving the track and then showing it at the next movement.

**Reply by silver-zepp** [reply]
Keep in mind our watches run C under the hood so our internal apps have access to more stuff that zos can provide. If you had 3.0 this would be solved by Canvas. 

Hmm, I have an experimental way of drawing arbitrary lines for 2.0, but it still won't be of any use for drawing curves but you can run them on an "unlimited" canvas (assuming you have unlimited RAM). I can share the implementation with you tomorrow. If you're planning on doing your own GPS app it should do it.
![gps-map-2 0](https://github.com/zepp-health/zeppos-forum/assets/138894963/988c2f77-4a76-47d6-bc27-51eb58a4040e)

**Reply by gekzakm** [reply]
Thank you, of course I would like to take advantage of your offer

**Reply by gekzakm** [reply]
![gps](https://github.com/zepp-health/zeppos-forum/assets/36265427/c1647fb9-4a52-4fda-a51c-9952bc6260d2)
I have stopped at this stage so far

**Reply by silver-zepp** [reply]
Done: https://github.com/silver-zepp/zeppos-draw. Have a look @gekzakm

**Reply by gekzakm** [reply]
Hi! In my opinion, this is what I need! Thanks for the support.

**Reply by gekzakm** [reply]
Hi. I'm asking for help.Tell me how to change the properties of an object created in a group? for example, I created a point (circle) and I need to change its location.
